### PR TITLE
SCAT-3622 fix for Author mapping

### PR DESCRIPTION
**Pull request by sudhakar-reddy-konda**
### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

### CODE REVIEW

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java [hunk 1]
```diff
@@ -382,20 +386,37 @@ private CaTMessage convertMessageToCatMessage(
...
+            .author(getAuthorDetails(message));
+  }
+
+  private CaTMessageOCDSAllOfAuthor getAuthorDetails(
+          final uk.gov.crowncommercial.dts.scale.cat.model.jaggaer.Message message) {
+    CaTMessageOCDSAllOfAuthor author;
+    if (CaTMessageNonOCDS.DirectionEnum.RECEIVED.getValue().equals(message.getDirection())) {
+      author = new CaTMessageOCDSAllOfAuthor()
+              .id(message.getSender().getId())
+              .name(message.getSender().getName());
```
Comment (adrianmilne) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java:407]: Have you applied the formatter to this Sudhakar? (if so still seems to be some issue with your formatting config)

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java [hunk 2]
```diff
@@ -382,20 +386,37 @@ private CaTMessage convertMessageToCatMessage(
...
+    } else {
+      var jaggaerUser = userProfileService.resolveBuyerUserByUserId(
+              String.valueOf(message.getSenderUser().getId()))
+              .orElseThrow(() -> new ResourceNotFoundException("Jaggaer"));
```
Comment (adrianmilne) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java:411]: Might want to consider adding the id to the error message to assist any troubleshooting down the line

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java (hunk 1 above)
Comment (sudhakar-reddy-konda) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java]: Please check now

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java (hunk 2 above)
Comment (adrianmilne) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java]: Might also want to apply same to the `Conclave` user error below

**Review by adrianmilne — APPROVED** [review]
👍

**Pull request merged**